src/Lib/Canvas.vala

```diff
@@ -153,70 +153,88 @@ public class Akira.Lib.Canvas : Goo.Canvas {
             motion_hover_event (event);
             return false;
         }
+        var canvas_event_x = event.x;
+        var canvas_event_y = event.y;
+        var event_x = canvas_event_x;
+        var event_y = canvas_event_y;
+
+        convert_to_item_space(selected_item, ref event_x, ref event_y);
+        var temp_event_x = canvas_temp_event_x;
+        var temp_event_y = canvas_temp_event_y;
+        convert_to_item_space(selected_item, ref temp_event_x, ref temp_event_y);
```

Comment:
wait a second, you might be having an extra variable here. canvas_temp_event_x was not changed before, so why make a new variable for these? and isn't `temp_event_x` also equal to `event_x`, so why two variables?

Reply:
It is at the end of this function. canvas_temp_event_x is the last position processed. Happens that rotate transform is not absolute, but cumulative, so the angle from canvas_event_x_root will not work. canvas_temp_event_x (and its item_space pair temp_event_x) allows to get the angle rotated from the very previous notify_motion_event.